Test fail..

Are some of the tests indeterminate?

I do

````
$ git clone git://github.com/node-opcua/node-opcua.git
$ cd node-opcua
$ npm install
$ npm test
````

and I now get (I am sure I wasn't before on same commit..)

````

1) TR8 -  disconnecting during connect


  886 passing (1m)
  4 pending
  1 failing

  1) testing ability for client to reconnect when server close connection TR8 -  disconnecting during connect:
     Uncaught AssertionError: expected 0 to be above 2
      at Assertion.fail (node_modules/should/cjs/should.js:231:17)
      at Assertion.value (node_modules/should/cjs/should.js:305:19)
      at test/end_to_end/test_e2e_connection_reconnection.js:1368:49
      at node_modules/async/dist/async.js:3686:9
      at replenish (node_modules/async/dist/async.js:882:17)
      at iterateeCallback (node_modules/async/dist/async.js:867:17)
      at node_modules/async/dist/async.js:840:16
      at node_modules/async/dist/async.js:3691:13
      at apply (node_modules/async/dist/async.js:21:25)
      at node_modules/async/dist/async.js:56:12
      at test/end_to_end/test_e2e_connection_reconnection.js:64:13
      at Timeout._onTimeout (test/end_to_end/test_e2e_connection_reconnection.js:714:13)



npm ERR! Test failed.  See above for more details.


````

--- a/lib/client/client_base.js
+++ b/lib/client/client_base.js
@@ -363,7 +363,7 @@ OPCUAClientBase.prototype._internal_create_secure_channel = function (callback) 
                     _inner_callback(err);
                 });
             } else {
-                // end point are already known
+                // end points are already known
                 _inner_callback(null);
             }
         }

--- a/test/end_to_end/test_e2e_connection_reconnection.js
+++ b/test/end_to_end/test_e2e_connection_reconnection.js
@@ -1298,7 +1298,7 @@ describe("testing ability for client to reconnect when server close connection",
         });
     });
 
-    xit("TR7 -  a client with some active monitoring items should be able to seamlessly reconnect after a very long connection break exceeding session life time", function (done) {
+    xit("TR7 -  a client with some active monitored items should be able to reconnect seamlessly after a very long connection break exceeding session life time", function (done) {
         // to do
         async.series([
 
@@ -1339,11 +1339,11 @@ describe("testing ability for client to reconnect when server close connection",
 
         var backoff_event_counter =0;
         client.on("backoff", function (err) {
-            // xx console.log("...");
+            console.log("...");
             backoff_event_counter+=1;
         });
 
-        var endpointUrl = "opc.tcp://somewhere-far-away.in.an.other.galaxy:42";
+        endpointUrl = "opc.tcp://somewhere-far-away.in.an.other.galaxy.com:4242";
 
         // let's call connect.
         // because the endpointUrl doesn't exist,  and the the infinite_connectivity_strategy
@@ -1353,42 +1353,48 @@ describe("testing ability for client to reconnect when server close connection",
         var connect_err = null;
         client.connect(endpointUrl, function(err) {
             connect_err = err;
+            //xx console.log("client.connect(err) err = ",err);
             connect_done = true;
         });
 
         var count_ref = 0;
 
         async.series([
-            f(wait_a_little_while),
-            f(wait_a_little_while),
+
+            // Wait until backoff is raised several times
+            function(callback) {
+                client.once("backoff", function(number,delay) { callback(); });
+            },
+            function(callback) {
+                client.once("backoff", function(number,delay) { callback(); });
+            },
+            function(callback) {
+                client.once("backoff", function(number,delay) { callback(); });
+            },
             function(callback) {
 
-                //  When client#disconnect is called
-
                 backoff_event_counter.should.be.greaterThan(2);
-
                 // client should be still trying to connect
                 connect_done.should.eql(false);
+
+                //  When client#disconnect is called
                 client_has_received_close_event.should.eql(0);
 
                 client.disconnect(function(err) {
 
-                    //xx setTimeout(function() {
-                        client_has_received_close_event.should.eql(1);
-                        // connect callback should have been called...
-                        connect_done.should.eql(true);
+                    client_has_received_close_event.should.eql(1);
+                    // connect callback should have been called...
+                    connect_done.should.eql(true);
 
-                        //xx console.log("-----------------------------------------------------");
-                        callback(err);
-                        count_ref = backoff_event_counter;
+                    callback(err);
+                    count_ref = backoff_event_counter;
 
-                    //xx },10);
                 });
             },
             f(wait_a_little_while),
             function(callback) {
                 client_has_received_close_event.should.eql(1);
-                // backof must be terminated now
+                // backoff must be terminated now
                 count_ref.should.eql(backoff_event_counter);
                 callback(null);
             }

